libavformat/movenccenc.c

```diff
@@ -83,21 +90,50 @@ static int auxiliary_info_add_subsample(MOVMuxCencContext* ctx,
     return 0;
 }
 
+void ff_mov_cenc_auxiliary_info_reset(MOVMuxCencContext* ctx) {
+    ctx->auxiliary_info_size = 0;
+    ctx->auxiliary_info_entries = 0;
+}
+
 /**
  * Encrypt the input buffer and write using avio_write
+ *
+ * For AES-CBC, the block chain starting with MOVMuxCencContext->aes_cbc_iv is
+ * wholly contained in this function; i.e. the chain does not span multiple
+ * calls.
  */
 static void mov_cenc_write_encrypted(MOVMuxCencContext* ctx, AVIOContext *pb,
                                      const uint8_t *buf_in, int size)
 {
-    uint8_t chunk[4096];
+    uint8_t chunk[4000];
```

Comment:
Is there a reason to change 4096 to 4000?  Maybe also add a comment as to what this means

Reply:
Ah ... it's kind of documented by the assert. This is so every call to `av_aes_crypt` is a multiple of 160, instead of 16, because the pattern repeats is every 10 blocks.

Speaking of which, the pattern is hard-coded to 1/9 encrypted/clear ratio (recommended by the spec I think). Not sure if there is a reason to encrypt more.